src/Sarif.Converters/ContrastSecurityConverter.cs

```diff
@@ -579,6 +577,11 @@ private Result ConstructFormsAuthenticationSSLResult(ContrastLogReader.Context c
 
         private Result ConstructFormsWithoutAutocompletePreventionResult(ContrastLogReader.Context context)
         {
+            // The maximum length of a snippet that Contrast Security embeds in their HTML file.
+            // If the actual <form> element is longer than this, they tack on an ellipsis, which
+            // we don't want to include in the SARIF file.
+            const int MaxSnippetLength = 1000;
```

Comment:
>const int MaxSnippetLength = 1000; [](start = 12, length = 34)

does it make sense that this be moved to a private constant at the class level? I know its only being used in this method today, this is mostly for furutre: 

- would be easier to spot and update in future.
- used in other functions if needed.

Reply:
It's a subtle question. The two bullet points are distinct:

To your first point : If this constant continues to be used _only in this method_, and it needs to be updated, then there will be no problem finding it _in this method_.

The real concern is your second point: What if I find I need a similar constant in a different method? Will I remember that it already exists in this method?

To be honest, the answer is "probably not". So the trade-off is between current understandability and future-proofing. There's this principle that says you should reduce the scope of symbols as much as possible to reduce "cognitive load" on the reader. Suppose I define this symbol at the top of the file (500 lines earlier). Then someone reading the file from the top has to hold this symbol in their heads, even though they might never read the one method that uses it. OTOH, someone who reads just this method and encounters a _usage_ of the symbol has to jump _out_ of the method to find out what it means.

So there is definitely a trade-off, but I almost always come down on the side of reduced scope, unless it is clear to me that we're going to use it elsewhere in the immediate future. It's just another example of the YAGNI principle.

---
In reply to: [294483628](https://github.com/microsoft/sarif-sdk/pull/1535#discussion_r294483628) [](ancestors = 294483628)